diff --git a/core/modules/node/src/Controller/NodeController.php b/core/modules/node/src/Controller/NodeController.php
index 87c9586d..0986b0c2 100644
--- a/core/modules/node/src/Controller/NodeController.php
+++ b/core/modules/node/src/Controller/NodeController.php
@@ -210,6 +210,31 @@ public function revisionOverview(NodeInterface $node) {
               '#suffix' => '</em>',
             ],
           ];
+          
+          // Add operations for current revision if it is not the latest.
+          $links = [];
+          if ($revision->access('revert revision') && !$revision->isLatestRevision()) {
+            $links['revert'] = [
+              'title' => $this->t('Copy as latest revision'),
+              'url' => $has_translations
+                ? Url::fromRoute('node.revision_revert_translation_confirm', [
+                    'node' => $node->id(),
+                    'node_revision' => $vid,
+                    'langcode' => $langcode,
+                  ])
+                : Url::fromRoute('node.revision_revert_confirm', [
+                    'node' => $node->id(),
+                    'node_revision' => $vid,
+                  ]),
+            ];
+          }
+
+          $row[] = [
+            'data' => [
+              '#type' => 'operations',
+              '#links' => $links,
+            ],
+          ];
 
           $rows[] = [
             'data' => $row,
diff --git a/core/modules/node/src/NodeAccessControlHandler.php b/core/modules/node/src/NodeAccessControlHandler.php
index 13020cea..f016aa94 100644
--- a/core/modules/node/src/NodeAccessControlHandler.php
+++ b/core/modules/node/src/NodeAccessControlHandler.php
@@ -163,7 +163,7 @@ protected function checkAccess(EntityInterface $node, $operation, AccountInterfa
       // If this is the default revision, return access denied for revert or
       // delete operations.
       $cacheability->addCacheableDependency($node);
-      if ($node->isDefaultRevision() && ($operation === 'revert revision' || $operation === 'delete revision')) {
+      if ($node->isDefaultRevision() && $operation === 'delete revision') {
         return AccessResult::forbidden()->addCacheableDependency($cacheability);
       }
       elseif ($account->hasPermission('administer nodes')) {
