diff --git a/src/anchorediting.js b/src/anchorediting.js
--- a/src/anchorediting.js
+++ b/src/anchorediting.js
@@ -103,10 +103,6 @@ export default class AnchorEditing extends Plugin {
 					value: viewElement => {
 						if ( viewElement.childCount < 1 ) {
 							return;
-						}
-
-						if ( viewElement.hasAttribute( 'href' ) ) {
-							return;
 						}
 
 						return viewElement.getAttribute( 'id' );