diff --git a/core/modules/ckeditor5/src/Controller/CKEditor5MediaController.php b/core/modules/ckeditor5/src/Controller/CKEditor5MediaController.php
index a725510d6b..391a48efce 100644
--- a/core/modules/ckeditor5/src/Controller/CKEditor5MediaController.php
+++ b/core/modules/ckeditor5/src/Controller/CKEditor5MediaController.php
@@ -11,6 +11,7 @@
 use Drupal\Core\Entity\EntityRepositoryInterface;
 use Drupal\Core\Session\AccountInterface;
 use Drupal\editor\Entity\Editor;
+use Drupal\file\Entity\File;
 use Drupal\image\Plugin\Field\FieldType\ImageItem;
 use Drupal\media\MediaInterface;
 use Symfony\Component\DependencyInjection\ContainerInterface;
@@ -116,6 +117,14 @@ public function mediaEntityMetadata(Request $request) {
       }
     }

+    // Adding the URL string to edit media items to metadata.
+    $response['edit_url'] = $media->toUrl('edit-form')->toString();
+
+    // Adding file URL to image source metadata.
+    $file_id = $media->{$image_field}->target_id;
+    $file = File::load($file_id);
+    $response['imageSourceMetadata']['filepath'] = $file->createFileUrl();
+
     // Note that we intentionally do not use:
     // - \Drupal\Core\Cache\CacheableResponse because caching it on the server
     //   side is wasteful, hence there is no need for cacheability metadata.
