diff --git a/src/DESConnector/Elasticsearch/Aggregations/Aggregations.php b/src/DESConnector/Elasticsearch/Aggregations/Aggregations.php
index f0c3d0b..5e7835c 100644
--- a/src/DESConnector/Elasticsearch/Aggregations/Aggregations.php
+++ b/src/DESConnector/Elasticsearch/Aggregations/Aggregations.php
@@ -65,7 +65,7 @@ class Aggregations implements AggregationsInterface
      *
      * @return void
      */
-    private function __wakeup()
+    public function __wakeup()
     {
     }
 
