diff --git a/config/schema/environment_indicator.schema.yml b/config/schema/environment_indicator.schema.yml
index c6197ca..79202e3 100644
--- a/config/schema/environment_indicator.schema.yml
+++ b/config/schema/environment_indicator.schema.yml
@@ -19,6 +19,9 @@ environment_indicator.switcher.*:
     url:
       type: uri
       label: 'URL'
+    path:
+      type: string
+      label: 'Path'
     fg_color:
       type: string
       label: 'Foreground color'
diff --git a/environment_indicator.module b/environment_indicator.module
index 86e8163..3e84963 100644
--- a/environment_indicator.module
+++ b/environment_indicator.module
@@ -14,7 +14,8 @@ use Drupal\environment_indicator\Entity\EnvironmentIndicator;
  * Implements hook_help().
  */
 function environment_indicator_help($route_name, RouteMatchInterface $route_match) {
-  $permissions_url = Url::fromRoute('user.admin_permissions', [], ['fragment' => 'module-environment_indicator'])->toString();
+  $permissions_url = Url::fromRoute('user.admin_permissions', [], ['fragment' => 'module-environment_indicator'])
+    ->toString();
   $settings_url = Url::fromRoute('environment_indicator.settings')->toString();
   switch ($route_name) {
     case 'environment_indicator.settings':
@@ -68,7 +69,8 @@ function environment_indicator_page_top(array &$page_top) {
     '#title' => $title,
     '#fg_color' => $active_environment->get('fg_color'),
     '#bg_color' => $active_environment->get('bg_color'),
-    '#description' => \Drupal::state()->get('environment_indicator.current_release'),
+    '#description' => \Drupal::state()
+      ->get('environment_indicator.current_release'),
     '#access' => !empty($title) && \Drupal::currentUser()
         ->hasPermission('access environment indicator'),
     '#attached' => [
@@ -205,7 +207,8 @@ function environment_indicator_toolbar() {
   if (!_environment_indicator_external_integration_is_enabled('toolbar')) {
     return $items;
   }
-  $release_info = \Drupal::state()->get('environment_indicator.current_release');
+  $release_info = \Drupal::state()
+    ->get('environment_indicator.current_release');
   if (!empty($release_info)) {
     $release_info = '(' . $release_info . ')';
   }
@@ -292,7 +295,7 @@ function _environment_indicator_switcher_links() {
     return [];
   }
   $current = Url::fromRoute('<current>');
-  $current_path = $current->toString();
+  $remote_path = $current->toString();
   $environment_entities = array_filter(
     $environment_entities,
     function (EnvironmentIndicator $entity) {
@@ -300,14 +303,26 @@ function _environment_indicator_switcher_links() {
     }
   );
   return array_map(
-    function (EnvironmentIndicator $entity) use ($current_path) {
+    function (EnvironmentIndicator $entity) use ($remote_path) {
+      $title = t('Opens the current page in the selected environment.');
+      $link_text = t('Open in: @label', ['@label' => $entity->label()]);
+      if ($path = $entity->getPath()) {
+        if ($path == '<front>') {
+          $remote_path = '';
+        }
+        else {
+          $remote_path = $path;
+        }
+        $title = t('Switch to the selected environment.');
+        $link_text = t('Open: @label', ['@label' => $entity->label()]);
+      }
       return [
         'attributes' => [
           'style' => 'color: ' . $entity->getFgColor() . '; background-color: ' . $entity->getBgColor() . ';',
-          'title' => t('Opens the current page in the selected environment.'),
+          'title' => $title,
         ],
-        'title' => t('Open in: @label', ['@label' => $entity->label()]),
-        'url' => Url::fromUri($entity->getUrl() . $current_path),
+        'title' => $link_text,
+        'url' => Url::fromUri($entity->getUrl() . $remote_path),
         'type' => 'link',
         'weight' => $entity->getWeight(),
       ];
@@ -333,7 +348,8 @@ function _environment_indicator_switcher_links() {
 function _environment_indicator_external_integration_is_enabled($integration) {
   if ($integration == 'toolbar') {
     $access = \Drupal::moduleHandler()->moduleExists('toolbar');
-    $toolbar_integration = \Drupal::config('environment_indicator.settings')->get('toolbar_integration') ?? [];
+    $toolbar_integration = \Drupal::config('environment_indicator.settings')
+      ->get('toolbar_integration') ?? [];
     $access = $access && in_array('toolbar', $toolbar_integration);
     $access = $access && \Drupal::currentUser()
         ->hasPermission('access toolbar');
diff --git a/src/Entity/EnvironmentIndicator.php b/src/Entity/EnvironmentIndicator.php
index b06ab40..dae292e 100644
--- a/src/Entity/EnvironmentIndicator.php
+++ b/src/Entity/EnvironmentIndicator.php
@@ -13,8 +13,10 @@ use Drupal\Core\Config\Entity\ConfigEntityInterface;
  *   label = @Translation("Environment Switcher"),
  *   handlers = {
  *     "storage" = "Drupal\Core\Config\Entity\ConfigEntityStorage",
- *     "access" = "Drupal\environment_indicator\EnvironmentIndicatorAccessControlHandler",
- *     "list_builder" = "Drupal\environment_indicator\EnvironmentIndicatorListBuilder",
+ *     "access" =
+ *   "Drupal\environment_indicator\EnvironmentIndicatorAccessControlHandler",
+ *     "list_builder" =
+ *   "Drupal\environment_indicator\EnvironmentIndicatorListBuilder",
  *     "form" = {
  *       "default" = "Drupal\environment_indicator\EnvironmentIndicatorForm",
  *       "delete" = "Drupal\Core\Entity\EntityDeleteForm"
@@ -29,9 +31,11 @@ use Drupal\Core\Config\Entity\ConfigEntityInterface;
  *     "weight" = "weight"
  *   },
  *   links = {
- *     "edit-form" = "/admin/config/development/environment-indicator/manage/{environment_indicator}",
+ *     "edit-form" =
+ *   "/admin/config/development/environment-indicator/manage/{environment_indicator}",
  *     "edit-permissions-form" = "/admin/people/permissions/{user_role}",
- *     "delete-form" = "/admin/config/development/environment-indicator/manage/{environment_indicator}/delete",
+ *     "delete-form" =
+ *   "/admin/config/development/environment-indicator/manage/{environment_indicator}/delete",
  *     "collection" = "/admin/config/development/environment-indicator"
  *   },
  *   config_export = {
@@ -39,6 +43,7 @@ use Drupal\Core\Config\Entity\ConfigEntityInterface;
  *     "description",
  *     "name",
  *     "url",
+ *     "path",
  *     "weight",
  *     "fg_color",
  *     "bg_color",
@@ -68,6 +73,14 @@ class EnvironmentIndicator extends ConfigEntityBase implements ConfigEntityInter
    */
   public $url;
 
+  /**
+   * The path to switch to on the environment. If empty, switch to same path as
+   * current environment.
+   *
+   * @var string
+   */
+  public $path;
+
   /**
    * The switchers link weight.
    *
@@ -113,6 +126,17 @@ class EnvironmentIndicator extends ConfigEntityBase implements ConfigEntityInter
     return $this->get('url');
   }
 
+  /**
+   * Gets the path.
+   *
+   * @return string
+   *   The path to switch to on the environment. If empty, switch to same path
+   *   as current environment.
+   */
+  public function getPath() {
+    return $this->get('path');
+  }
+
   /**
    * Gets the weight.
    *
@@ -183,6 +207,16 @@ class EnvironmentIndicator extends ConfigEntityBase implements ConfigEntityInter
     $this->set('url', $url);
   }
 
+  /**
+   * Sets the path.
+   *
+   * @param string $url
+   *   The path.
+   */
+  public function setPath($path) {
+    $this->set('path', $path);
+  }
+
   /**
    * Sets the foreground color.
    *
diff --git a/src/EnvironmentIndicatorForm.php b/src/EnvironmentIndicatorForm.php
index 26b017d..39a15d2 100644
--- a/src/EnvironmentIndicatorForm.php
+++ b/src/EnvironmentIndicatorForm.php
@@ -44,6 +44,16 @@ class EnvironmentIndicatorForm extends EntityForm {
       '#default_value' => $environment_switcher->getWeight(),
     ];
 
+    $form['path'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Path'),
+      '#description' => $this->t('Optionally set a static path to switch
+      to on this environment (ex: /user/login). Leave empty to link the user to
+      the same path they are viewing on the current environment. Use
+      &lt;front&gt; to link to the homepage.'),
+      '#default_value' => $environment_switcher->getPath(),
+    ];
+
     $form['bg_color'] = [
       '#type' => 'color',
       '#title' => $this->t('Background Color'),
