diff --git a/core/modules/ckeditor5/js/ckeditor5_plugins/drupalMedia/src/drupalmediaui.js b/core/modules/ckeditor5/js/ckeditor5_plugins/drupalMedia/src/drupalmediaui.js
index ccef719d..a02465f6 100644
--- a/core/modules/ckeditor5/js/ckeditor5_plugins/drupalMedia/src/drupalmediaui.js
+++ b/core/modules/ckeditor5/js/ckeditor5_plugins/drupalMedia/src/drupalmediaui.js
@@ -29,7 +29,7 @@ export default class DrupalMediaUI extends Plugin {
       const buttonView = new ButtonView(locale);
 
       buttonView.set({
-        label: Drupal.t('Insert Media'),
+        label: Drupal.t('Embed images and other media files'),
         icon: mediaIcon,
         tooltip: true,
       });
