diff --git a/core/modules/layout_builder/src/Plugin/Derivative/ExtraFieldBlockDeriver.php b/core/modules/layout_builder/src/Plugin/Derivative/ExtraFieldBlockDeriver.php
index 087ef13d85e..87af3993473 100644
--- a/core/modules/layout_builder/src/Plugin/Derivative/ExtraFieldBlockDeriver.php
+++ b/core/modules/layout_builder/src/Plugin/Derivative/ExtraFieldBlockDeriver.php
@@ -101,6 +101,10 @@ public function getDerivativeDefinitions($base_plugin_definition) {
     $enabled_bundle_ids = $this->bundleIdsWithLayoutBuilderDisplays();
     $expose_all_fields = $this->moduleHandler->moduleExists('layout_builder_expose_all_field_blocks');
     foreach ($this->entityTypeManager->getDefinitions() as $entity_type_id => $entity_type) {
+      // Don't process webform_submission entities.
+      if ($entity_type_id == 'webform_submission') {
+        continue;
+      }
       // Only process fieldable entity types.
       if (!$entity_type->entityClassImplements(FieldableEntityInterface::class)) {
         continue;
diff --git a/core/modules/layout_builder/src/Plugin/Derivative/FieldBlockDeriver.php b/core/modules/layout_builder/src/Plugin/Derivative/FieldBlockDeriver.php
index e01a0300201..b75962a4f4e 100644
--- a/core/modules/layout_builder/src/Plugin/Derivative/FieldBlockDeriver.php
+++ b/core/modules/layout_builder/src/Plugin/Derivative/FieldBlockDeriver.php
@@ -107,6 +107,10 @@ public static function create(ContainerInterface $container, $base_plugin_id) {
   public function getDerivativeDefinitions($base_plugin_definition) {
     $entity_type_labels = $this->entityTypeRepository->getEntityTypeLabels();
     foreach ($this->getFieldMap() as $entity_type_id => $entity_field_map) {
+      // Don't process webform_submission entities.
+      if ($entity_type_id == 'webform_submission') {
+        continue;
+      }
       foreach ($entity_field_map as $field_name => $field_info) {
         // Skip fields without any formatters.
         $options = $this->formatterManager->getOptions($field_info['type']);
