diff --git a/media_migration.module b/media_migration.module
index 8bb0d31..e2d896f 100644
--- a/media_migration.module
+++ b/media_migration.module
@@ -4,10 +4,3 @@
  * @file
  * Primary module hooks for Media migration module.
  */
-
-/**
- * Implements hook_migration_plugins_alter().
- */
-function media_migration_migration_plugins_alter(array &$migrations) {
-  \Drupal::service('media_migration.migrate_plugin_alterer')->alter($migrations);
-}
