diff --git a/src/SamlService.php b/src/SamlService.php
index e093d75..6a1e0a9 100644
--- a/src/SamlService.php
+++ b/src/SamlService.php
@@ -584,7 +584,7 @@ class SamlService {
       }
     }
     elseif ($account->isBlocked()) {
-      throw new UserVisibleException('Requested account is blocked.');
+      throw new UserVisibleException('Requested account is blocked. Contact WebCMS Support.');
     }
     else {
       // Synchronize the user account with SAML attributes if needed.
