diff --git a/js/views/WidgetView.js b/js/views/WidgetView.js
index da6ac0aba370eb37566d3e6addee9d3a0a14ff48..33bc92a2012f0dbcecf35d74152b507c4e2a2943 100644
--- a/js/views/WidgetView.js
+++ b/js/views/WidgetView.js
@@ -66,6 +66,10 @@
               // Add core class to apply default styles to the element.
               .addClass('form-select')
               .hide();
+      if (widget.model.itemCollection.length > 0) {
+        widget.$el.attr('aria-label', widget.container.app.$el.siblings('label').text());
+        widget.container.app.$el.siblings('label').prop('for', widget.container.app.$el.prop('id') + '-shs-' + widget.container.model.get('delta') + '-' + widget.model.get('level'));
+      }
       if (widget.model.get('dataLoaded')) {
         widget.$el.show();
       }
