diff --git a/src/Plugin/Field/FieldWidget/EntityGroupFieldWidgetBase.php b/src/Plugin/Field/FieldWidget/EntityGroupFieldWidgetBase.php
index efac04d..d6e8d41 100644
--- a/src/Plugin/Field/FieldWidget/EntityGroupFieldWidgetBase.php
+++ b/src/Plugin/Field/FieldWidget/EntityGroupFieldWidgetBase.php
@@ -235,6 +235,12 @@ abstract class EntityGroupFieldWidgetBase extends WidgetBase implements Containe
       $element['#prefix'] = '<div id="' . $wrapper_id . '">';
       $element['#suffix'] = '</div>';
 
+      // To support group_media we need to check the $entity_plugin_id as the
+      // permissions change based on that. For example group_media uses
+      // 'update own group_media:image entity' compared to what other permissions
+      // use, i.e. 'update own group_node:page content'.
+      $suffix = str_contains($entity_plugin_id, 'group_media') ? 'entity' : 'content';
+
       // Check permissions.
       if ($entity_plugin_id == 'group_membership') {
         if ($items->getEntity()->id() == $account->id()) {
@@ -247,8 +253,8 @@ abstract class EntityGroupFieldWidgetBase extends WidgetBase implements Containe
         }
       }
       else {
-        $can_delete = $host->isNew() ? FALSE : $group->hasPermission("delete any $entity_plugin_id content", $account);
-        $can_edit = $group->hasPermission("update any $entity_plugin_id content", $account);
+        $can_delete = $host->isNew() ? FALSE : $group->hasPermission("delete any $entity_plugin_id $suffix", $account);
+        $can_edit = $group->hasPermission("update any $entity_plugin_id $suffix", $account);
       }
       // Checking if can delete own.
       if (!$can_delete && $entity_plugin_id == 'group_membership') {
@@ -257,7 +263,7 @@ abstract class EntityGroupFieldWidgetBase extends WidgetBase implements Containe
         }
       }
       if (!$can_delete && $gcontent_entity->getOwnerId() == $account->id()) {
-        $can_delete = $group->hasPermission("delete own $entity_plugin_id content", $account);
+        $can_delete = $group->hasPermission("delete own $entity_plugin_id $suffix", $account);
       }
       // Checking if can update own.
       if (!$can_edit) {
@@ -268,7 +274,7 @@ abstract class EntityGroupFieldWidgetBase extends WidgetBase implements Containe
           $group_content_owner = $gcontent_entity->getEntity()->id();
         }
         if ($group_content_owner == $account->id()) {
-          $can_edit = $group->hasPermission("update own $entity_plugin_id content", $account);
+          $can_edit = $group->hasPermission("update own $entity_plugin_id $suffix", $account);
         }
       }
 
