diff --git a/modules/tome_static/src/StaticGenerator.php b/modules/tome_static/src/StaticGenerator.php
index 845568f..41a5d51 100644
--- a/modules/tome_static/src/StaticGenerator.php
+++ b/modules/tome_static/src/StaticGenerator.php
@@ -167,6 +167,17 @@ class StaticGenerator implements StaticGeneratorInterface {
     }
 
     $destination = $this->getDestination($path);
+    if ($response->isRedirection()) {
+      preg_match('/(sites\/default\/files|s3\/files)/i', $response->getTargetUrl(), $matches);
+      if (!empty($matches)) {
+        $redirection_url = $this->makeExternalUrlLocal($response->getTargetUrl());
+        if (!UrlHelper::isExternal($redirection_url)) {
+          $redirection_path = str_replace($matches[0], '', ltrim(parse_url($redirection_url, PHP_URL_PATH), '/'));
+          $redirection_uri = 'public://' . ltrim(rawurldecode(rawurldecode($redirection_path)), '/');
+          $response = new BinaryFileResponse($redirection_uri, 200);
+        }
+      }
+    }
     if ($response->isRedirection() || $response->isOk()) {
       $directory = dirname($destination);
       $this->fileSystem->prepareDirectory($directory, FileSystemInterface::CREATE_DIRECTORY);
