diff --git a/modules/tome_static/src/StaticGenerator.php b/modules/tome_static/src/StaticGenerator.php
index 845568f..d2fbaf0 100644
--- a/modules/tome_static/src/StaticGenerator.php
+++ b/modules/tome_static/src/StaticGenerator.php
@@ -23,6 +23,7 @@ use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\RequestStack;
 use Symfony\Component\HttpKernel\HttpKernelInterface;
 use Drupal\Core\File\FileSystemInterface;
+use Drupal\Core\File\FileUrlGeneratorInterface;
 
 /**
  * Handles static site generation.
@@ -82,6 +83,13 @@ class StaticGenerator implements StaticGeneratorInterface {
    */
   protected $requestStack;
 
+  /**
+   * The file URL generator.
+   *
+   * @var \Drupal\Core\File\FileUrlGeneratorInterface
+   */
+  protected $fileUrlGenerator;
+
   /**
    * Creates a StaticGenerator object.
    *
@@ -97,8 +105,10 @@ class StaticGenerator implements StaticGeneratorInterface {
    *   The account switcher.
    * @param \Drupal\Core\File\FileSystemInterface $file_system
    *   The file system.
+   * @param \Drupal\Core\File\FileUrlGeneratorInterface $file_url_generator
+   *   The file URL generator.
    */
-  public function __construct(HttpKernelInterface $http_kernel, RequestStack $request_stack, EventDispatcherInterface $event_dispatcher, StaticCacheInterface $cache, AccountSwitcherInterface $account_switcher, FileSystemInterface $file_system) {
+  public function __construct(HttpKernelInterface $http_kernel, RequestStack $request_stack, EventDispatcherInterface $event_dispatcher, StaticCacheInterface $cache, AccountSwitcherInterface $account_switcher, FileSystemInterface $file_system, FileUrlGeneratorInterface $file_url_generator) {
     $this->httpKernel = $http_kernel;
     $this->currentRequest = $request_stack->getCurrentRequest();
     $this->eventDispatcher = $event_dispatcher;
@@ -106,7 +116,7 @@ class StaticGenerator implements StaticGeneratorInterface {
     $this->accountSwitcher = $account_switcher;
     $this->requestStack = $request_stack;
     $this->fileSystem = $file_system;
-
+    $this->fileUrlGenerator = $file_url_generator;
   }
 
   /**
@@ -173,7 +183,7 @@ class StaticGenerator implements StaticGeneratorInterface {
       // This is probably an image style derivative.
       if ($response instanceof BinaryFileResponse) {
         $file_path = $response->getFile()->getPathname();
-        $this->copyPath($file_path, $destination);
+        $invoke_paths[] = $this->fileUrlGenerator->generateAbsoluteString($file_path);
       }
       else {
         $content = $response->getContent();
diff --git a/modules/tome_static/tome_static.services.yml b/modules/tome_static/tome_static.services.yml
index 921f2dd..610c43e 100644
--- a/modules/tome_static/tome_static.services.yml
+++ b/modules/tome_static/tome_static.services.yml
@@ -1,7 +1,7 @@
 services:
   tome_static.generator:
     class: Drupal\tome_static\StaticGenerator
-    arguments: ['@http_kernel', '@request_stack', '@event_dispatcher', '@cache.tome_static', '@account_switcher', '@file_system']
+    arguments: ['@http_kernel', '@request_stack', '@event_dispatcher', '@cache.tome_static', '@account_switcher', '@file_system', '@file_url_generator']
   cache.tome_static:
     class: Drupal\tome_static\StaticCache
     arguments: ['@database', '@cache_tags.invalidator.checksum']
