diff --git a/src/GroupMenuService.php b/src/GroupMenuService.php
index a24d6d7..313c0a0 100644
--- a/src/GroupMenuService.php
+++ b/src/GroupMenuService.php
@@ -197,11 +197,11 @@ class GroupMenuService implements GroupMenuServiceInterface {
         foreach ($group_contents as $group_content) {
           // Make sure the group and entity IDs are set in the group content
           // entity.
-          if (!isset($group_content->gid->target_id) || !isset($group_content->entity_id->target_id)) {
+          if (!isset($group_content->gid->target_id) || !isset($group_content->{$group_content->getEntityFieldName()}->target_id)) {
             continue;
           }
           /** @var \Drupal\group\Entity\GroupContentInterface $group_content */
-          $this->groupMenus[$group_content->gid->target_id][$group_content->entity_id->target_id] = $menus[$group_content->entity_id->target_id];
+          $this->groupMenus[$group_content->gid->target_id][$group_content->{$group_content->getEntityFieldName()}->target_id] = $menus[$group_content->{$group_content->getEntityFieldName()}->target_id];
         }
       }
     }
